README.md

```diff
@@ -1,8 +1,113 @@
 # Cloud Spanner Banking Application
 
-This repository contains a sample banking application built using Cloud Spanner.
+This repository provides the source code for a Cloud Spanner sample financial
+application. It consists of 2 components:
+1. A gRPC server that provides application level functionality
+  (see: [service.proto](server/src/main/proto/service.proto)) using CloudSpanner
+  as the storage backend, found in the [server](server) directory
+2. A workload generator for the above, found in the [workload](workload) folder
 
-## How to run the application against Cloud Spanner emulator
+The goal of the sample is to provide a simple working example to jump-start
+exploring Cloud Spanner.
+
+## Architecture & Schema
+
+### Application Components
+This is how everything fits together:
+![FinAppConnect.png](images/FinAppConnect.png)
+
+The FinAppServer is the backend of the application. It provides a gRPC interface
+with application-level operations and uses Cloud Spanner as the storage backend.
+We provide two alternative implementations for connecting to Cloud Spanner for
+educational purposes:
+- [Cloud Spanner Java Client](https://github.com/googleapis/java-spanner)
+- [Cloud Spanner JDBC](https://github.com/googleapis/java-spanner-jdbc)
+
+Clients, including the workload generator, a potential frontend component, or
+tools like [grpc_cli](https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md)
+can connect to the FinAppServer using gRPC.
+
+### ER Diagram & Operations
+
+The application models the following entities:
+- *Customer*: End-user of the application. A customer can have one or more
+  accounts.
+- *Account*: A financial account, keeps track of balance. Multiple customers
+  could access a single account with different roles.
+- *Transaction*: A monetary transfers associated with the accounts
+
+Captured as an ER diagram:
+![FinAppERD.png](images/FinAppERD.png)
+
+The backend should support the following operations:
+- *CreateCustomer*
+- *CreateAccount*
+- *CreateCustomerRole*: Associate a customer with a role for a specified
+  account
+- *CreateTransactionForAccount*: Monetary transfer between an account and an
+  external entity
+- *MoveAccountBalance*: Monetary transfer between two accounts
+- *GetRecentTransactionsForAccount*: Returns recent transactions for an account,
+  should support pagination.
```

Comment:
Can you provide information on what defines 'recent'? Is it configurable?

Reply:
Added more details.